orca_detector/vggish_model.py

```diff
@@ -0,0 +1,261 @@
+"""
+VGGish model for Keras. A VGG-like model for audio classification
+
+W251 (Summer 2019) - Spyros Garyfallos, Ram Iyer, Mike Winton
+
+Adapted from https://github.com/DTaoo/VGGish
+"""
+
+import numpy as np
+import sys
+
+from keras.models import Model
+from keras.layers import Flatten, Dense, Input, Conv2D, MaxPooling2D, GlobalAveragePooling2D, GlobalMaxPooling2D
+from keras.engine.topology import get_source_inputs
+from keras import backend as K
+
+# project-specific imports
+import orca_params
+import vggish_params as params
+
+# weight path; when Docker container is run, weights path on the host
+# machine is expected to be mapped to /weights
+WEIGHTS_PATH = '/weights/vggish_audioset_weights_without_fc2.h5'
+WEIGHTS_PATH_TOP = '/weights/vggish_audioset_weights.h5'
+
+class OrcaVGGish(object):    
+    """
+    An adaption of the VGGish architecture for our audio classification model.
+    This model is never trained if weights aren't provided!
+    """
+    
+    def __init__ (self,
+                  load_weights=True, 
+                  weights='audioset', 
+                  input_tensor=None,
+                  input_shape=None,
+                  out_dim=None,
+                  pooling='avg'):
+
+        """
+        :param load_weights: if load weights
+        :param weights: loads weights pre-trained on a preliminary version of YouTube-8M.
+        :param input_tensor: input_layer
+        :param input_shape: input data shape
+        :param out_dim: output dimension
+        :param pooling: pooling type over the non-top network, 'avg' or 'max'
+
+        :return: A Keras model instance.
+        """
+
+        # Validate parameters
+        if weights not in {'audioset', None}:
+            raise ValueError('The `weights` argument should be either '
+                             '`None` (random initialization) or `audioset` '
+                             '(pre-training on audioset).')
+
+        out_dim = orca_params.NUM_CLASSES
+
+        if input_shape is None:
+            input_shape = (params.NUM_FRAMES, params.NUM_BANDS, 1)  # 496, 64, 1
+
+        # VGGish model was trained with a "batch-first" matrix format.
+        if input_tensor is None:
+            aud_input = Input(shape=input_shape, name='input_1')
```

Comment:
Nit: please rename aud_input to audio_input

Reply:
done